generator/src/main/java/org/corfudb/generator/Generator.java

```diff
@@ -19,55 +19,196 @@
  * patterns can be approximated by setting different configurations like
  * data distributions, concurrency level, operation types, time skews etc.
  *
- *
  * Created by maithem on 7/14/17.
+ *
+ * Modified by Miao Cheng on 10/15/18:
+ *
+ * Re-implemented the Generator class to provide APIs for running workloads that can be
+ * 'divided' into chunks, of which transactions are decoupled from each other.
+ * This can help do correctness verification for Longevity test chunk-by-chunk on the fly.
  */
+@Slf4j
 public class Generator {
-    public static void main(String[] args) {
-        String endPoint = args[0];
-        int numStreams = Integer.parseInt(args[1]);
-        int numKeys = Integer.parseInt(args[2]);
-        long cpPeriod = Long.parseLong(args[3]);
-        int numThreads = Integer.parseInt(args[4]);
-        int numOperations = Integer.parseInt(args[5]);
 
-        CorfuRuntime rt = new CorfuRuntime(endPoint).connect();
+    private long cpPeriod;
+    private Duration totalRuntime;
+
+    static final int numStreams = 50;
+    static final int numKeys = 100;
+    static final int numThreads = 10;
+
+    BlockingQueue<Operation> operationQueue;
+    CorfuRuntime rt;
+    State state;
+    ScheduledExecutorService scheduler;
+    ExecutorService workers;
+
+    // The lock prevents consuming the queue while it's being filled.
+    Boolean operationQueueLock;
+
+    // Given the above configuration (numStreams, numKeys, numThreads),
+    // this queue capacity can be consumed for approximately 3 minutes.
+    static final int QUEUE_CAPACITY = 100000;
+
+    static final Duration CHUNK_RUNTIME = Duration.ofHours(1);
+    static final Duration CHUNK_TIMEOUT = Duration.ofSeconds(5);
+    static final Duration POLL_TIME = Duration.ofSeconds(3);
+    static final Duration LIVENESS_TIMEOUT = Duration.ofMinutes(15);
+    static final long TIME_TO_WAIT_FOR_RUNTIME_TO_CONNECT = 60000;
+
+
+    public Generator(String endPoint, long cpPeriod, Duration totalRuntime) {
+
+        this.cpPeriod = cpPeriod;
+        this.totalRuntime = totalRuntime;
+
+        rt = new CorfuRuntime(endPoint).connect();
+
+        try {
+            tryToConnectTimeout(TIME_TO_WAIT_FOR_RUNTIME_TO_CONNECT);
```

Comment:
Is runtime already connected at line 65?

Reply:
oh, forgot to remove the connect() from the line that created the runtime, will fix this.